### PR TITLE
Feature/115 add convenience metho

### LINKED ISSUE #115: Add convenience method to add driver
Add a method that allows users to add and register a JDBC driver.

Currently, we have to do this by writing the following lines to `drivers/jdbc/settings.cfg:
```
DRIVERNAME=MYSQL
JAR=mysql-connector-java-8.0.20.jar
DRIVERMAIN=com.mysql.jdbc.Driver
PREFIX=jdbc:mysql:
NOSECURITY=YES
FETCHSIZE=100000
INSERTSIZE=-1
```

and upload the driver to `drivers/jdbc/`.

Suggested API:

```java
EXASOL = new ExasolContainer();
EXASOL.addJdbcDriver(file, name, prefix, disableSecurtity);
```

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #115.

### CODE REVIEW

File: doc/system_requirements.md [hunk 1]
```diff
@@ -400,6 +407,27 @@ Covers:
 
 Needs: dsn
 
+### Driver Management
+
+The Exasol database supports installation of drivers for external data sources through [BucketFS](#bucketfs-access). Installation requires Integrators to follow a convention for storing and registering those drivers.
```
Comment (jakobbraun) [inline doc/system_requirements.md:412]: is there public documentation on that? In that case you could link it here.

File: src/main/java/com/exasol/drivers/DriverManagerException.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.exasol.drivers;
+
+public class DriverManagerException extends RuntimeException {
```
Comment (jakobbraun) [inline src/main/java/com/exasol/drivers/DriverManagerException.java:3]: javadoc is missing

File: src/main/java/com/exasol/drivers/ExasolDriverManager.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            + MANIFEST_FILENAME;
+    private final Map<String, DatabaseDriver> installedDrivers = new HashMap<>();
+    private final Bucket bucket;
+
+    /**
+     * Create a new instance of an {@link ExasolDriverManager}.
+     *
+     * @param bucket BucketFS bucket in which the drivers should be installed
+     */
+    public ExasolDriverManager(final Bucket bucket) {
```
Comment (jakobbraun) [inline src/main/java/com/exasol/drivers/ExasolDriverManager.java:29]: So we should be able to extract the `ExasolDriverManager` with the bucketfs code one day. :+1:

File: src/main/java/com/exasol/drivers/JdbcDriver.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+         * @return new JDBC driver
+         */
+        public JdbcDriver build() {
+            validate();
+            return new JdbcDriver(this);
+        }
+
+        private void validate() {
+            if ((this.name == null) || !this.name.matches("[a-zA-Z]\\w+")) {
+                throw new IllegalStateException("Empty or illegal driver name (" + this.name
```
Comment (jakobbraun) [inline src/main/java/com/exasol/drivers/JdbcDriver.java:160]: use error codes

File: src/test/java/com/exasol/drivers/ExasolDriverManagerIT.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                .mainClass("org.example.DummyDriver").build();
+        driverManager.install(driver);
+        final Bucket bucket = EXASOL.getDefaultBucket();
+        assertAll(
+                () -> assertThat("Driver file in Bucket",
+                        bucket.isObjectSynchronized(ExasolDriverManager.DEFAULT_JDBC_DRIVER_PATH_IN_BUCKET
+                                + BucketConstants.PATH_SEPARATOR + fileName, beforeInstallation),
+                        equalTo(true)),
+                () -> assertThat("Manifest file in Bucket", bucket.isObjectSynchronized(
+                        ExasolDriverManager.DEFAULT_JDBC_DRIVER_PATH_IN_BUCKET, beforeInstallation), equalTo(true)));
```
Comment (jakobbraun) [inline src/test/java/com/exasol/drivers/ExasolDriverManagerIT.java:50]: isen't the file name missing here?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: doc/system_requirements.md (hunk 1 above)
Comment (redcatbear) [reply doc/system_requirements.md:412]: Added.

File: src/main/java/com/exasol/drivers/DriverManagerException.java (hunk 2 above)
Comment (redcatbear) [reply src/main/java/com/exasol/drivers/DriverManagerException.java:3]: Added.

File: src/main/java/com/exasol/drivers/ExasolDriverManager.java (hunk 3 above)
Comment (redcatbear) [reply src/main/java/com/exasol/drivers/ExasolDriverManager.java:29]: That was part of the plan. Although it is a separate concern that just happens to have a Bucket dependency.

File: src/main/java/com/exasol/drivers/JdbcDriver.java (hunk 4 above)
Comment (redcatbear) [reply src/main/java/com/exasol/drivers/JdbcDriver.java:160]: Next PR. This one is already too big.

File: src/test/java/com/exasol/drivers/ExasolDriverManagerIT.java (hunk 5 above)
Comment (redcatbear) [reply src/test/java/com/exasol/drivers/ExasolDriverManagerIT.java:50]: That is intentional. I was thinking of cases where the file is already in BFS and only needs to be registered.

On the other hand maybe that is such a fringe case that we could simply ignore it.

**Review by jakobbraun — APPROVED** [review]

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**